File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashMap<Character,Integer> h=new HashMap<>();
        int max=0,l=0,i;
        for(i=0;i<s.length();i++){
            if(!h.containsKey(s.charAt(i)))
               h.put(s.charAt(i),i); 
            else {
                if(max<i-l)max=i-l;
                l=Math.max(h.get(s.charAt(i))+1,l);
                h.put(s.charAt(i),i); 
            }
        }
        if(max<i-l)max=i-l;
        return max;
    }
}

File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char,int> um;
        int maxi=0,l=0,i;
        for(i=0;s[i]!='\0';i++){
            if(!um.count(s[i]))
               um[s[i]]=i; 
            else {
                if(maxi<i-l)maxi=i-l;
                l=max(um[s[i]]+1,l);
                um[s[i]]=i; 
            }
        }
        if(maxi<i-l)maxi=i-l;
        return maxi;
    }
};